src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/NewAzureRmVpnClientIpsecParametersCommand.cs

```diff
@@ -0,0 +1,141 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using System;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, "AzureRmVpnClientIpsecParameter", SupportsShouldProcess = true), OutputType(typeof(PSVpnClientIPsecParameters))]
+    public class NewAzureRmVpnClientIpsecParametersCommand : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient IPSec Security Association (also called Quick Mode or Phase 2 SA) lifetime in seconds")]
+        [ValidateRange(300, 172799)]
+        public int SALifeTime { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient IPSec Security Association (also called Quick Mode or Phase 2 SA) payload size in KB")]
+        [ValidateRange(1024, int.MaxValue)]
+        public int SADataSize { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient IPSec encryption algorithm (IKE Phase 1)")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IpsecEncryption.GCMAES256,
+            MNM.IpsecEncryption.GCMAES128,
+            MNM.IpsecEncryption.AES256,
+            MNM.IpsecEncryption.AES128,
+            IgnoreCase = false)]
```

Comment:
@Nilambari is there any reason we need to be case sensitive for all of these parameters using `ValidateSet`? I think we should make all `ValidateSet` attributes case insensitive, so removing the `IgnoreCase = False` line from each one will resolve this issue.